Optimizer/SimpleOptimizer/SimpleOptimizer.cpp

```diff
@@ -151,37 +152,40 @@ void SimpleOptimizer::optimize(std::vector<std::shared_ptr<Organism>> &populatio
 	  }
   }
   aveScore /= oldPopulationSize;
-
-  auto checkScores = scores;
   
   if (cullBelow >= 0 && minScore != maxScore){ // cull and normalize scores if min == max then all scores are the same, do nothing!
+	culledMinScore = maxScore;
+	culledMaxScore = maxScore;
 	cullBelowScore = minScore + ((maxScore-minScore) * cullBelow);
-	std::cout << "\n\nmax: " << maxScore << "   min: " << minScore;
+	//std::cout << "\n\nmax: " << maxScore << "   min: " << minScore;
 	deltaScore = maxScore - cullBelowScore;
-	std::cout << "  cullBelowScore: " << cullBelowScore << "  deltaScore: " << deltaScore << std::endl;
+	//std::cout << "  cullBelowScore: " << cullBelowScore << "  deltaScore: " << deltaScore << std::endl;
 	for (int i = 0; i < (int)population.size(); i++) {
```

Comment:
We need to replace all instances of c-style casts, with the appropriate c++ cast; which in most cases is simply a static_cast. I have this in my list of things to change in rewrite, but I haven't gotten to all of them yet (precisely because one of the reasons c-style casts are bad, is that they are painful to search for). PRs are definitely a good place to start. '
I'm aware this seems pedantic, but just google "c++ c-style casts", and you'll see what I mean.

Reply:
Googled (from a SO post) because I couldn't remember the reasons either:
"
- C++ style casts are checked by the compiler. C style casts aren't and can fail at runtime
- also, c++ style casts can be searched for easily, whereas it's really hard to search for c style casts
- Another big benefit is that the 4 different C++ style casts express the intent of the programmer more clearly. "